evalml/data_checks/null_data_check.py

```diff
@@ -178,9 +178,19 @@ def validate(self, X, y=None):
         }
 
         X = infer_feature_types(X)
+        X_to_validate = X.ww.select(
```

Comment:
Updating this data check to only check categorical + numerical cols (+ unknown which might just be fully null cols).

Reason for this is because two-fold:
1. We have specific data checks such as the `DateTimeNaNDataCheck` data check and the `NaturalLanguageNaNDataCheck` data check for specific types.
2. https://github.com/alteryx/evalml/issues/3241: PerColumnImputer does not currently support datetime columns, so if we have a nan value that gets passed along as a parameter, then we'll error out 😬

Reply:
Open to discussion though! Seems like we assume natural language columns will be included, given `test_default_data_checks_classification`

Reply:
I thought it was confusing that `NaturalLanguage` is included in `X_to_validate` given your comment that this data check would only check "categorical + numerical cols". 

My thought is that if we allow NaturalLanguage & DateTime in this check and update the action for those columns to only be "drop column" if they are highly null, then we can close out #3240 and #3242 with minimal changes to what you have here. 

Since this is going into the feature branch, I think that change can be made in a follow-up pr though.